ext/openssl/openssl.c

```diff
@@ -4228,180 +4228,280 @@ static EVP_PKEY *php_openssl_pkey_init_dh(zval *data, bool *is_private)
 #ifdef HAVE_EVP_PKEY_EC
 #if PHP_OPENSSL_API_VERSION < 0x30000
 static bool php_openssl_pkey_init_legacy_ec(EC_KEY *eckey, zval *data, bool *is_private) {
+	BIGNUM *p = NULL, *a = NULL, *b = NULL, *g_x = NULL, *g_y = NULL, *order = NULL, *cofactor = NULL;
+	BIGNUM *x = NULL, *y = NULL, *d = NULL;
 	EC_GROUP *group = NULL;
-	EC_POINT *pnt = NULL;
-	BIGNUM *d = NULL;
-	zval *bn;
-	zval *x;
-	zval *y;
+	EC_POINT *point_g = NULL;
+	EC_POINT *point_q = NULL;
+	BN_CTX *bctx = BN_CTX_new();
+	zval *curve_name_zv = zend_hash_str_find(Z_ARRVAL_P(data), "curve_name", sizeof("curve_name") - 1);
+	zval *generator_zv = zend_hash_str_find(Z_ARRVAL_P(data), "generator", sizeof("generator") - 1);
+	zval *seed_zv = zend_hash_str_find(Z_ARRVAL_P(data), "seed", sizeof("seed") - 1);
```

Comment:
do not fetch those before they are actually needed - there is no point to do that if curve_name is provided so it should be done in the right branch before they are used.